crypto/checksum.ts

```diff
@@ -0,0 +1,39 @@
+import { crypto, hex, iterateReader } from '../deps.ts';
+
+type DigestAlgorithm = Parameters<typeof crypto.subtle.digest>[0];
+
+/**
+ * Computes the hash for the specified file.
+ *
+ * @param filePath The file to be hashed
+ * @param algorithm The digest algorithm to use when computing the hash
+ * @returns A {@link string} containing the hash in hexadecimal format
+ */
+export async function fileChecksumHex(
+  filePath: string,
+  algorithm: DigestAlgorithm = 'SHA-256',
+): Promise<string> {
+  const file = await Deno.open(filePath, { read: true });
+  try {
+    return await checksumHex(iterateReader(file), algorithm);
```

Comment:
The await here is superfluous since there is no next line. `checksumHex(...)` returns a `Promise<string>` so we should be solid.

```suggestion
    return checksumHex(iterateReader(file), algorithm);
```

Reply:
It's actually necessary here because the `finally` clause still needs to execute after it. If we close the file before reading is finished, it will break. I've added a comment pointing this out.

Reply:
Ah, that makes sense! TIL—thanks!